libbeat/idxmgmt/ilm/client_handler.go

```diff
@@ -139,110 +106,21 @@ func (h *ESClientHandler) HasILMPolicy(name string) (bool, error) {
-	status, b, err = h.client.Request("HEAD", "/"+name, "", nil, nil)
-	if err != nil && status != 404 {
-		return false, wrapErrf(err, ErrRequestFailed,
-			"failed to check for alias '%v': (status=%v) %s", name, status, b)
-	}
-	if status == 200 {
-		return false, errf(ErrInvalidAlias,
-			"resource '%v' exists, but it is not an alias", name)
-	}
-	return false, nil
-}
-
-// CreateAlias sends request to Elasticsearch for creating alias.
-func (h *ESClientHandler) CreateAlias(alias Alias) error {
-	// Escaping because of date pattern
-	// This always assume it's a date pattern by sourrounding it by <...>
-	firstIndex := fmt.Sprintf("<%s-%s>", alias.Name, alias.Pattern)
-	firstIndex = url.PathEscape(firstIndex)
-
-	body := common.MapStr{
-		"aliases": common.MapStr{
-			alias.Name: common.MapStr{
-				"is_write_index": true,
-			},
-		},
-	}
-
-	// Note: actual aliases are accessible via the index
-	if _, res, err := h.client.Request("PUT", "/"+firstIndex, "", nil, body); err != nil {
-		// Creating the index may fail for multiple reasons, e.g. because
-		// the index exists, or because the write alias exists and points
-		// to another index.
-		if ok, err := h.HasAlias(alias.Name); err != nil {
-			// HasAlias fails if there is an index with the same name.
-			return err
-		} else if ok {
-			return errOf(ErrAliasAlreadyExists)
-		}
-		return wrapErrf(err, ErrAliasCreateFailed, "failed to create alias: %s", res)
-	}
-	return nil
-}
-
-func (h *ESClientHandler) checkILMSupport() (avail, enabled bool, err error) {
-	var response struct {
-		Features struct {
-			ILM struct {
-				Available bool `json:"available"`
-				Enabled   bool `json:"enabled"`
-			} `json:"ilm"`
-		} `json:"features"`
-	}
-	status, err := h.queryFeatures(&response)
-	if status == 400 {
-		// If we get a 400, it's assumed to be the OSS version of Elasticsearch
-		return false, false, nil
-	}
-	if err != nil {
-		return false, false, wrapErr(err, ErrILMCheckRequestFailed)
-	}
-
-	avail = response.Features.ILM.Available
-	enabled = response.Features.ILM.Enabled
-	return avail, enabled, nil
-}
-
-func (h *ESClientHandler) queryFeatures(to interface{}) (int, error) {
-	status, body, err := h.client.Request("GET", esFeaturesPath, "", nil, nil)
-	if status >= 400 || err != nil {
-		return status, err
-	}
-
-	if to != nil {
-		if err := json.Unmarshal(body, to); err != nil {
-			return status, wrapErrf(err, ErrInvalidResponse, "failed to parse JSON response")
-		}
-	}
-	return status, nil
-}
-
 // CheckILMEnabled indicates whether or not ILM is supported for the configured mode and client version.
 func (h *FileClientHandler) CheckILMEnabled(enabled bool) (bool, error) {
+	return checkILMEnabled(enabled, h.client)
+}
+
+func checkILMEnabled(enabled bool, c VersionCheckerClient) (bool, error) {
 	if !enabled {
 		return false, nil
 	}
-	version := h.client.GetVersion()
-	if checkILMVersion(version) {
-		return enabled, nil
+
+	ver := c.GetVersion()
```

Comment:
Beats 8.0 should not work with ES 7.x in the first place. If we enforce this earlier in the chain, we can get rid of this too.

Reply:
I will open a follow up PR for this.

Reply:
Maybe this can be combined with https://github.com/elastic/beats/issues/29147